utils/check_docstrings.py

```diff
@@ -943,14 +944,27 @@ def fix_docstring(obj: Any, old_doc_args: str, new_doc_args: str):
         f.write("\n".join(lines))
 
 
-def check_docstrings(overwrite: bool = False):
+def check_docstrings(overwrite: bool = False, check_all: bool = False):
     """
-    Check docstrings of all public objects that are callables and are documented.
+    Check docstrings of all public objects that are callables and are documented. By default, only checks the diff.
 
     Args:
         overwrite (`bool`, *optional*, defaults to `False`):
             Whether to fix inconsistencies or not.
+        check_all (`bool`, *optional*, defaults to `False`):
+            Whether to check all files.
     """
+    module_diff_files = None
+    if not check_all:
+        module_diff_files = subprocess.run(
+            ["git", "diff", "--name-only", "origin/main"], capture_output=True, text=True
+        )
+        module_diff_files = module_diff_files.stdout.split("\n")
+        module_diff_files = {file for file in module_diff_files if file.startswith("src/transformers")}
```

Comment:
Let's use a git repo object here e.g. like in the tests_fetcher - it's cleaner than using a subprocess to exit out of python

Reply:
Done ✅